src/herder/Upgrades.cpp

```diff
@@ -1388,14 +1388,34 @@ ConfigUpgradeSetFrame::upgradeNeeded(LedgerSnapshot const& ls) const
 }
 
 void
-ConfigUpgradeSetFrame::applyTo(AbstractLedgerTxn& ltx) const
+ConfigUpgradeSetFrame::applyTo(AbstractLedgerTxn& ltx, Application& app) const
 {
+    bool writeLiveSorobanStateSizeWindow = false;
     for (auto const& updatedEntry : mConfigUpgradeSet.updatedEntry)
     {
         LedgerKey key(LedgerEntryType::CONFIG_SETTING);
         auto const id = updatedEntry.configSettingID();
         key.configSetting().configSettingID = id;
-        ltx.load(key).current().data.configSetting() = updatedEntry;
+        auto& currentEntry = ltx.load(key).current().data.configSetting();
+        if (currentEntry.configSettingID() ==
+                ConfigSettingID::CONFIG_SETTING_STATE_ARCHIVAL &&
+            currentEntry.stateArchivalSettings()
+                    .liveSorobanStateSizeWindowSampleSize !=
+                updatedEntry.stateArchivalSettings()
+                    .liveSorobanStateSizeWindowSampleSize)
+        {
+            writeLiveSorobanStateSizeWindow = true;
+        }
+        currentEntry = updatedEntry;
+    }
+
+    if (writeLiveSorobanStateSizeWindow)
+    {
+        SorobanNetworkConfig networkConfig;
+        networkConfig.loadFromLedger(
```

Comment:
This does some additional/unnecessary work, but the perf hit shouldn't matter here, and this allows us to just reuse existing code.